packages/devtools_app/lib/src/vm_developer/isolate_statistics_view.dart

```diff
@@ -0,0 +1,544 @@
+      child: VMInfoCard(
+        title: 'General',
+        rowKeyValues: [
+          MapEntry('Name', isolate?.name),
+          MapEntry('Started at', _startTime(isolate)),
+          MapEntry('Uptime', _uptime(isolate)),
+          MapEntry('Root Library', isolate?.rootLib?.uri),
+          MapEntry('ID', isolate?.id),
+        ],
+      ),
+    );
+  }
+}
+
+/// Displays memory statistics for the isolate, including:
+///   - Total Dart heap size
+///   - New + Old space capacity and usage
+///   - Handle counts (zone + scoped)
+///   - Per-thread zone memory usage
+class IsolateMemoryStatisticsWidget extends StatelessWidget {
+  const IsolateMemoryStatisticsWidget({@required this.controller});
+
+  final IsolateStatisticsViewController controller;
+
+  String _buildMemoryString(num usage, num capacity) {
+    if (usage == null || capacity == null) {
+      return '--';
+    }
+    return '${prettyPrintBytes(usage, includeUnit: true)}'
+        ' of ${prettyPrintBytes(capacity, includeUnit: true)}';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final isolate = controller.isolate;
+    return Column(
+      children: [
+        Flexible(
+          child: VMInfoCard(
+            title: 'Memory',
+            rowKeyValues: [
+              MapEntry(
+                'Dart Heap',
+                _buildMemoryString(
+                  isolate?.dartHeapSize,
+                  isolate?.dartHeapCapacity,
+                ),
+              ),
+              MapEntry(
+                'New Space',
+                _buildMemoryString(
+                  isolate?.newSpaceUsage,
+                  isolate?.newSpaceUsage,
+                ),
+              ),
+              MapEntry(
+                'Old Space',
+                _buildMemoryString(
+                  isolate?.oldSpaceUsage,
+                  isolate?.oldSpaceCapacity,
+                ),
+              ),
+              MapEntry(
+                'Max Zone Capacity',
+                prettyPrintBytes(
+                  controller.zoneCapacityHighWatermark,
+                  includeUnit: true,
+                ),
+              ),
+              MapEntry('# of Zone Handles', isolate?.zoneHandleCount),
+              MapEntry('# of Scoped Handles', isolate?.scopedHandleCount),
+            ],
+            table: Flexible(
+              child: ThreadMemoryTable(
+                controller: controller,
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// A table which displays the amount of time the VM is performing certain
+/// tagged tasks.
+class TagStatisticsWidget extends StatelessWidget {
+  TagStatisticsWidget({@required this.controller});
+
+  final name = _TagColumn();
+  final percentage = _PercentageColumn();
+
+  List<ColumnData<VMTag>> get columns => [name, percentage];
+  final IsolateStatisticsViewController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Flexible(
```

Comment:
instead of making the Flexible part of this widget, put the flexible directly inside the flex widget (row / column) where it is used. This makes the code more flexible (no pun intended) because as is, the TagStatisticsWidget can only be used inside flex widgets like Row or Column, but this is not evident unless you dig into the build method. Same comment for similar widgets like GeneralIsolateStatisticsWidget and ServiceExtensionsWidget. 

Placing the Flexible widgets directly in the Row or Column also makes the flex layout of the row more readable, as you can see what the flex values are directly in the flex widget where they are being used.